packages/pass-connector/src/connectors/abstract.ts

```diff
@@ -1,104 +1,118 @@
-import {Chain, Connector} from "@wagmi/core";
-import {checksumAddress, getAddress, http} from "viem";
-import {Passport} from "@0xpass/passport";
-import {createPassportClient} from "@0xpass/passport-viem";
-import {Address} from "wagmi";
-import {computeAddress} from "ethers";
-import {normalizeChainId} from "@0xpass/wagmi-commons";
-
-
-export const supportedProviders = [
-    // "apple",
-    // "bitbucket",
-    // "discord",
-    // "facebook",
-    // "github",
-    // "gitlab",
-    "google",
-    // "linkedin",
-    // "microsoft",
-    // "twitch",
-    // "twitter",
-] as const;
+import { Chain, Connector } from "@wagmi/core";
+import { getAddress, http } from "viem";
+import { Passport, Network } from "@0xpass/passport";
+import { WebauthnSigner } from "@0xpass/webauthn-signer";
+import { createPassportClient } from "@0xpass/passport-viem";
+import { Address } from "wagmi";
+import { normalizeChainId } from "@0xpass/wagmi-commons";
+import { AuthenticatedHeaders } from "@0xpass/passport/dist/types";
 
 const IS_SERVER = typeof window === "undefined";
 
 export interface PassOptions {
-    clientId: string;
-    scope: string;
-    verifier: typeof supportedProviders[number];
-    loginFn: () => Promise<any>;
+  scopeId: string;
+  network?: Network;
+  enclavePublicKey?: string;
+  username: string;
+  userDisplayName: string;
+  rpName: string;
+  rpId: string;
 }
 
-export abstract class AbstractPassConnector extends Connector<any,PassOptions> {
-    ready = !IS_SERVER;
-    readonly id = "pass";
-    readonly name = "Pass";
-    protected passport = new Passport()
-    protected sessionSig: undefined | string;
-    protected provider: undefined | any;
-    protected address: undefined | `0x${string}`;
-    protected chain: Chain
-
-    protected constructor(config: { chains?: Chain[]; options: PassOptions }) {
-        super(config);
-        this.chain = this.chains[0]
+export abstract class AbstractPassConnector extends Connector<
+  any,
+  PassOptions
+> {
+  ready = !IS_SERVER;
+  readonly id = "pass";
+  readonly name = "Pass";
+  protected passport;
+  protected authenticatedHeaders: undefined | AuthenticatedHeaders;
+  protected provider: undefined | any;
+  protected address: undefined | `0x${string}`;
+  protected chain: Chain;
+
+  protected constructor(config: { chains?: Chain[]; options: PassOptions }) {
+    super(config);
+    this.chain = this.chains[0];
+
+    this.passport = new Passport({
+      scopeId: this.options.scopeId,
+      signer: new WebauthnSigner({
+        rpName: this.options.rpName,
+        rpId: this.options.rpId,
```

Comment:
can we segregate this signer.... think there should be a way to set aprams from outside using PassOptions

Reply:
but u may run into async issues, so only do if test works otherwise chuck it or do it later